src/MSRIOGroup.cpp

```diff
@@ -732,6 +732,27 @@ namespace geopm
         return result;
     }
 
+    int MSRIOGroup::signal_behavior(const std::string &signal_name) const
+    {
+        if (!is_valid_signal(signal_name)) {
+            throw Exception("MSRIOGroup::signal_behavior(): signal_name " + signal_name +
+                            " not valid for MSRIOGroup",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
```

Comment:
Does every MSR field have an associated behavior defined?  If yes, then this should read something like:
```
MSRIOGroup::signal_behavior(): signal_name " + signal_name +
" is missing behavior field.",
```

Reply:
This is checked in the JSON parsing step for any MSRs/fields in the JSON files (see `check_msr_field()`).  This exception here should not really be possible to hit through PlatformIO because PlatformIO checks the available signal names before forwarding to the IOGroup.  This exception would only happen if a user managed to create an MSRIOGroup standalone and pass it an invalid name.  There is a similar exception for all the other methods that take signal names.

Reply:
Opened #1495